src/main/java/pl/javastart/streamsexercise/PaymentService.java

```diff
@@ -15,95 +21,177 @@ class PaymentService {
+        return getStream()
+                .filter(getPredicateForTwoDates(now, past))
+                .toList();
     }
 
+    private static Predicate<Payment> getPredicateForTwoDates(ZonedDateTime now, ZonedDateTime past) {
+        return payment -> payment.getPaymentDate().isBefore(now)
+                && payment.getPaymentDate().isAfter(past);
+    }
+
+
     /*
     Znajdź i zwróć płatności z jednym elementem
      */
     Set<Payment> findPaymentsWithOnePaymentItem() {
-        throw new RuntimeException("Not implemented");
+        return getStream()
+                .filter(PaymentService::checkIfOnlyOnePayment)
+                .collect(Collectors.toSet());
+    }
+
+    private static boolean checkIfOnlyOnePayment(Payment payment) {
+        return getPaymentSize(payment) == 1;
     }
 
     /*
     Znajdź i zwróć nazwy produktów sprzedanych w aktualnym miesiącu
      */
+
     Set<String> findProductsSoldInCurrentMonth() {
-        throw new RuntimeException("Not implemented");
+        List<Payment> paymentsForCurrentMonth = findPaymentsForCurrentMonth();
+        return paymentsForCurrentMonth.stream()
+                .map(Payment::getPaymentItems)
+                .flatMap(List::stream)
+                .map(PaymentItem::getName)
+                .collect(Collectors.toSet());
     }
-
     /*
     Policz i zwróć sumę sprzedaży dla wskazanego miesiąca
      */
+
     BigDecimal sumTotalForGivenMonth(YearMonth yearMonth) {
-        throw new RuntimeException("Not implemented");
+        return findPaymentsForGivenMonth(yearMonth).stream()
+                .map(Payment::countTotalSum)
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
     }
 
     /*
     Policz i zwróć sumę przyznanych rabatów dla wskazanego miesiąca
      */
+
     BigDecimal sumDiscountForGivenMonth(YearMonth yearMonth) {
-        throw new RuntimeException("Not implemented");
+        return findPaymentsForGivenMonth(yearMonth).stream()
+                .map(Payment::countDiscountSum)
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
     }
 
     /*
     Znajdź i zwróć płatności dla użytkownika z podanym mailem
      */
     List<PaymentItem> getPaymentsForUserWithEmail(String userEmail) {
-        throw new RuntimeException("Not implemented");
+        return getStream()
+                .filter(getEmailPredicate(userEmail))
+                .map(Payment::getPaymentItems)
+                .flatMap(List::stream)
+                .toList();
+    }
+
+    private static Predicate<Payment> getEmailPredicate(String userEmail) {
+        return payment -> payment.getUser().getEmail().equals(userEmail);
     }
 
     /*
     Znajdź i zwróć płatności, których wartość przekracza wskazaną granicę
      */
     Set<Payment> findPaymentsWithValueOver(int value) {
-        throw new RuntimeException("Not implemented");
+        return getStream()
+                .filter(payment -> checkIfSumBiggerThanValue(value, payment))
```

Comment:
to niespecjalnie błąd, ale skoro we wszystkich innych przypadkach wydzieliłaś metodę dla predykatów, to może tutaj też by było warto zachować konsekwencję